### PR TITLE
Sr/19828819 org not showing in listing

### PR DESCRIPTION
**Pull request by sachinr**
First commit - [fixes #19834651][fixes #19828819] Add All organization filter; fix reporting filter
Second commit - Validate presence of name and raw_type when admin updates organization

### CODE REVIEW

File: app/controllers/admin/organizations_controller.rb [hunk 1]
```diff
@@ -166,9 +166,11 @@ class Admin::OrganizationsController < Admin::BaseController
 
     # show reporting orgs by default.
     def scope_organizations(filter)
-      if filter == 'Non-Reporting'
-        Organization.nonreporting
-      elsif allowed_filter?(params[:filter])
+      if filter == 'Non-Reporting' || filter == 'Reporting'
+        Organization.send(filter.gsub('-','').downcase.to_sym)
```
Comment (glennr) [inline app/controllers/admin/organizations_controller.rb]: why go to all the trouble of this complex code 
  "filter.gsub('-','').downcase.to_sym" 

when you only have two symbols... 
  :nonreporting / :reporting ?

i would use an if...else... and be explicit about the method (i.e. named_scope) you are calling on the Org model...

File: app/models/organization.rb [hunk 2]
```diff
@@ -265,6 +265,19 @@ class Organization < ActiveRecord::Base
     end
   end
 
+  def almost_valid?
```
Comment (glennr) [inline app/models/organization.rb]: remove as discussed

File: app/models/organization.rb [hunk 3]
```diff
@@ -327,6 +340,19 @@ class Organization < ActiveRecord::Base
         end
       end
     end
+
+    def errors_on(fields)
```
Comment (glennr) [inline app/models/organization.rb]: remove

File: spec/controllers/admin/organizations_controller_spec.rb [hunk 4]
```diff
@@ -78,6 +89,45 @@ describe Admin::OrganizationsController do
     end
   end
 
+  describe "update organization" do
```
Comment (glennr) [inline spec/controllers/admin/organizations_controller_spec.rb]: remove ?

**Comment by sachinr** [comment]
Rebased and fixed failing cukes

**Pull request merged**